src/common/shared/shared.c

```diff
@@ -1120,14 +1120,10 @@ Q_strlcpy(char *dst, const char *src, int size)
 {
 	const char *s = src;
 
-	while (*s)
+	while (--size > 0)
```

Comment:
fails for size=0

Reply:
(what's wrong with the original implementation anyway?)

Reply:
> fails for size=0

true even tough only used with compile time buffers it seems.

Reply:
> (what's wrong with the original implementation anyway?)

I realise ... is those functions trying to detect truncations ?

e.g.

`
char buf[5];
Q_strlcpy(buf, "abcdef", sizeof(buf));
`

will return 6. 

if that s the case I will revert the change.

Reply:
they're supposed to behave like https://www.freebsd.org/cgi/man.cgi?query=strlcpy&sektion=3

Reply:
clear enough